### PR TITLE
[ADD] Attendees validation in E,A,T

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

**Review by rimbendhaou — CHANGES_REQUESTED** [review]
Can you add unit tests ?

File: project_event/tests/test_project_event_task.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from datetime import datetime, timedelta
 from odoo import exceptions
 from odoo import fields
-from .common import TestProjectEventCommon
+from common import TestProjectEventCommon
```
Comment (eilst) [inline project_event/tests/test_project_event_task.py]:
```suggestion
from .common import TestProjectEventCommon
```

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -1012,3 +1027,16 @@ def get_confirmation_wizard(self, action):
...
+        vals = {
+            'start': self.date_start,
+            'stop': self.date_end,
+            'partner_ids': [(6, 0, [attendee])]}
+        v_event = self.env['calendar.event'].new(vals)
+        [(6, 0, self.get_partners())]
+        try:
+            v_event._check_resources_booked(v_event, v_event.partner_ids)
+        except Exception as e:
+            return [e.name, e.name.split(' ')[1], e.name.split(' ')[2]]
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:1041]: this traceback is displayed when clicking on option:
Traceback (most recent call last):
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 651, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 693, in dispatch
    result = self._call_function(**self.params)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 937, in __call__
    return self.method(*args, **kw)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/rbendhaou/Documents/Projets/Udem/git/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/rbendhaou/Documents/Projets/Udem/git/project-addons/project_event/models/project_project.py", line 111, in action_option
    return self.get_confirmation_wizard('option')
  File "/home/rbendhaou/Documents/Projets/Udem/git/project-addons/project_event/models/project_project.py", line 207, in get_confirmation_wizard
    res += activity.get_booked_resources()
  File "/home/rbendhaou/Documents/Projets/Udem/git/project-addons/project_event/models/project_task.py", line 724, in get_booked_resources
    hres = child.get_booked_attendees(attendee)
  File "/home/rbendhaou/Documents/Projets/Udem/git/project-addons/project_event/models/project_task.py", line 1041, in get_booked_attendees
    return [e.name, e.name.split(' ')[1], e.name.split(' ')[2]]
AttributeError: 'DataError' object has no attribute 'name'

**Review by rimbendhaou — APPROVED** [review]

**Review by lgharib — COMMENTED** [review]
**Test status:** Failed

**Steps to reproduce:**
Testing with two employee one Demo user (Allowed double booking = False) and David (Allowed double booking True).
Create Calendar event in Calendar module with both **Attendees**
Create Task in Event module with both **Employees**.
Change Task Event status to Option.

**Expected alert message:**
1# Please confirm your reservation.
The following resources are already booked:
**David
Demo**
Do you want to continue?
2# The attendee **David** cannot be double-booked with any overlapping meetings or events. 

**Actual message:**
1# Please confirm your reservation.
The following resources are already booked:
**David
Demo**
Do you want to continue?
2# The attendee **David** cannot be double-booked with any overlapping meetings or events.

**Comment by eilst** [comment]
@rimbendhaou @lgharib Can we merge this?

**Pull request merged**